Resource leak may occurred in RestfulService

## Bug Report

```
[ERROR] 2020-11-12 09:05:05,100 --nioEventLoopGroup-19-1-- [io.netty.util.ResourceLeakDetector] LEAK: ByteBuf.release() was not called before it's garbage-collected. See https://netty.io/wiki/reference-counted-objects.html for more information.
Recent access records: 
Created at:
	io.netty.buffer.AbstractByteBufAllocator.compositeDirectBuffer(AbstractByteBufAllocator.java:223)
	io.netty.buffer.AbstractByteBufAllocator.compositeBuffer(AbstractByteBufAllocator.java:201)
	io.netty.handler.codec.MessageAggregator.decode(MessageAggregator.java:271)
	io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:88)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
	io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355)
	io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
	io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:321)
	io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:295)
	io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
	io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355)
	io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
	io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
	io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	java.lang.Thread.run(Thread.java:748) 
```

--- a/elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/NettyRestfulService.java
+++ b/elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/NettyRestfulService.java
@@ -24,6 +24,7 @@ import io.netty.channel.EventLoopGroup;
 import io.netty.channel.nio.NioEventLoopGroup;
 import io.netty.channel.socket.nio.NioServerSocketChannel;
 import io.netty.util.NettyRuntime;
+import lombok.RequiredArgsConstructor;
 import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.shardingsphere.elasticjob.restful.pipeline.RestfulServiceChannelInitializer;
@@ -32,6 +33,7 @@ import org.apache.shardingsphere.elasticjob.restful.pipeline.RestfulServiceChann
  * Implemented {@link RestfulService} via Netty.
  */
 @Slf4j
+@RequiredArgsConstructor
 public final class NettyRestfulService implements RestfulService {
     
     private static final int DEFAULT_WORKER_GROUP_THREADS = 1 + 2 * NettyRuntime.availableProcessors();
@@ -43,10 +45,6 @@ public final class NettyRestfulService implements RestfulService {
     private EventLoopGroup bossEventLoopGroup;
     
     private EventLoopGroup workerEventLoopGroup;
-    
-    public NettyRestfulService(final NettyRestfulServiceConfiguration configuration) {
-        this.configuration = configuration;
-    }
     
     private void initServerBootstrap() {
         bossEventLoopGroup = new NioEventLoopGroup();

--- a/elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/pipeline/HandleMethodExecutor.java
+++ b/elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/pipeline/HandleMethodExecutor.java
@@ -27,6 +27,7 @@ import io.netty.handler.codec.http.HttpHeaderNames;
 import io.netty.handler.codec.http.HttpResponseStatus;
 import io.netty.handler.codec.http.HttpUtil;
 import io.netty.handler.codec.http.HttpVersion;
+import io.netty.util.ReferenceCountUtil;
 import org.apache.shardingsphere.elasticjob.restful.handler.HandleContext;
 import org.apache.shardingsphere.elasticjob.restful.handler.Handler;
 import org.apache.shardingsphere.elasticjob.restful.serializer.ResponseBodySerializer;
@@ -44,21 +45,23 @@ public final class HandleMethodExecutor extends ChannelInboundHandlerAdapter {
     @Override
     public void channelRead(final ChannelHandlerContext ctx, final Object msg) throws Exception {
         HandleContext<Handler> handleContext = (HandleContext<Handler>) msg;
-        Handler handler = handleContext.getMappingContext().payload();
-        Object[] args = handleContext.getArgs();
-        
-        Object handleResult = handler.execute(args);
-        
-        FullHttpResponse response;
-        if (null != handleResult) {
-            String mimeType = HttpUtil.getMimeType(handler.getProducing()).toString();
-            ResponseBodySerializer serializer = ResponseBodySerializerFactory.getResponseBodySerializer(mimeType);
-            byte[] bodyBytes = serializer.serialize(handleResult);
-            response = createHttpResponse(handler.getProducing(), bodyBytes, handler.getHttpStatusCode());
-        } else {
-            response = createHttpResponse(handler.getProducing(), new byte[0], handler.getHttpStatusCode());
+        try {
+            Handler handler = handleContext.getMappingContext().payload();
+            Object[] args = handleContext.getArgs();
+            Object handleResult = handler.execute(args);
+            FullHttpResponse response;
+            if (null != handleResult) {
+                String mimeType = HttpUtil.getMimeType(handler.getProducing()).toString();
+                ResponseBodySerializer serializer = ResponseBodySerializerFactory.getResponseBodySerializer(mimeType);
+                byte[] bodyBytes = serializer.serialize(handleResult);
+                response = createHttpResponse(handler.getProducing(), bodyBytes, handler.getHttpStatusCode());
+            } else {
+                response = createHttpResponse(handler.getProducing(), new byte[0], handler.getHttpStatusCode());
+            }
+            ctx.writeAndFlush(response);
+        } finally {
+            ReferenceCountUtil.release(handleContext.getHttpRequest());
         }
-        ctx.writeAndFlush(response);
     }
     
     private FullHttpResponse createHttpResponse(final String producingContentType, final byte[] bodyBytes, final int statusCode) {

--- a/elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/pipeline/HttpRequestDispatcher.java
+++ b/elasticjob-infra/elasticjob-restful/src/main/java/org/apache/shardingsphere/elasticjob/restful/pipeline/HttpRequestDispatcher.java
@@ -22,6 +22,7 @@ import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.ChannelInboundHandlerAdapter;
 import io.netty.handler.codec.http.FullHttpRequest;
 import io.netty.handler.codec.http.HttpMethod;
+import io.netty.util.ReferenceCountUtil;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.shardingsphere.elasticjob.restful.handler.HandleContext;
 import org.apache.shardingsphere.elasticjob.restful.handler.Handler;
@@ -64,6 +65,7 @@ public final class HttpRequestDispatcher extends ChannelInboundHandlerAdapter {
         }
         MappingContext<Handler> mappingContext = mappingRegistry.getMappingContext(request);
         if (null == mappingContext) {
+            ReferenceCountUtil.release(request);
             throw new HandlerNotFoundException(request.uri());
         }
         HandleContext<Handler> handleContext = new HandleContext<>(request, mappingContext);

